### PR TITLE
feat: Implement authorized chars and conversion rules for node id field - EXO-63217 - Meeds-io/meeds#51

### PR DESCRIPTION
**Pull request by IlhemEssaadi**

### CODE REVIEW

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue [hunk 1]
```diff
@@ -132,22 +137,54 @@ export default {
...
         required: value => (value == null || value.length) || this.$t('siteNavigation.required.error.message'),
       },
+      nodeIdRules: [
+        value => {
+          if (value !== null && (/\s+/.test(value) || /[^a-zA-Z0-9_-]/.test(value) || /[\u0300-\u036f]/.test(value.normalize('NFD')))){
+            return this.$t('siteNavigation.UnauthorizedCharacters.error.message');
+          } else {
+            return (value == null || value.length) || this.$t('siteNavigation.required.error.message');
+          }
+
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue]:
```suggestion
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue [hunk 2]
```diff
@@ -132,22 +137,54 @@ export default {
...
+    }
+  },
   methods: {
-    open() {
+    open(navigationNode) {
+      const siteKey = navigationNode.siteKey;
+      if (siteKey.typeName === 'portal') {
+        this.nodeUrl = `/portal/${siteKey.name}/${navigationNode.uri}`;
+      } else if (siteKey.typeName === 'group' && siteKey.name.includes('spaces')) {
+        this.nodeUrl = `/portal/g/${siteKey.name.replaceAll('/', ':')}`;
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue]: useless condition IMO, a spaces node or a group node has the same uri

**Review by azayati — CHANGES_REQUESTED** [review]

File: layout-management-webapps/src/main/resources/locale/portlet/layoutManagement/SiteNavigationPortlet_en.properties [hunk 3]
```diff
@@ -21,14 +21,16 @@ siteNavigation.label.moveDown=Move down
...
+siteNavigation.label.nodeId.title=Node id
+siteNavigation.label.nodeId.description=Display in the url:
 siteNavigation.label.visibility.title=Visibility
 siteNavigation.label.visibility.visible=Visible node
 siteNavigation.label.visibility.scheduleVisibility=Schedule the node's visibility?
 siteNavigation.label.nodeType.title=Node type
 siteNavigation.label.nodeType.group=Group
 siteNavigation.label.nodeType.pageOrLink=Page or link
 siteNavigation.required.error.message=This field is required
+siteNavigation.UnauthorizedCharacters.error.message=Only lowercase characters, uppercase characters, _, -, and numbers are authorized
```
Comment (azayati) [inline layout-management-webapps/src/main/resources/locale/portlet/layoutManagement/SiteNavigationPortlet_en.properties]:
```suggestion
siteNavigation.unauthorizedCharacters.error.message=Only lowercase characters, uppercase characters, _, -, and numbers are authorized
```

**Review by azayati — CHANGES_REQUESTED** [review]

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue [hunk 4]
```diff
@@ -132,22 +137,52 @@ export default {
...
+        }
+      ],
     };
   },
   created() {
     this.$root.$on('open-site-navigation-add-node-drawer', this.open);
   },
+  watch: {
+    nodeId() {
+      this.nodeUrl = `${this.InitialUrl }/${this.nodeId}`;
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue]:
```suggestion
      this.nodeUrl = `${this.InitialUrl}/${this.nodeId}`;
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue [hunk 5]
```diff
@@ -132,22 +137,52 @@ export default {
...
     close() {
       this.nodeId = null;
       this.nodeLabel = null;
       this.$refs.siteNavigationAddNodeDrawer.close();
+    },
+    conversionRules() {
+      return this.nodeLabel.normalize('NFD').replace(/[\u0300-\u036f]/g, '').replace(/[^a-zA-Z0-9_-]/g, '').replace(/\s+/g, '').toLowerCase();
+    },
+    blurOnNodeLabel() {
+      if (this.nodeId == null){
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue]:
```suggestion
      if (this.nodeId == null) {
```

**Review by azayati — CHANGES_REQUESTED** [review]

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue [hunk 6]
```diff
@@ -132,22 +137,52 @@ export default {
...
     this.$root.$on('open-site-navigation-add-node-drawer', this.open);
   },
+  watch: {
+    nodeId() {
+      this.nodeUrl = `${this.InitialUrl}/${this.nodeId}`;
+    }
+  },
   methods: {
-    open() {
+    open(navigationNode) {
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue]:
```suggestion
    open(parentNavigationNode) {
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue [hunk 7]
```diff
@@ -132,22 +137,52 @@ export default {
...
+      const siteKey = navigationNode.siteKey;
+      if (siteKey.name === 'dw') {
+        this.InitialUrl = `/portal/${siteKey.name}/${navigationNode.uri}`;
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue]:
```suggestion
        this.parentNavigationNodeUrl = `/portal/${siteKey.name}/${navigationNode.uri}`;
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue [hunk 8]
```diff
@@ -132,22 +137,52 @@ export default {
...
+      } else {
+        this.InitialUrl = `/portal/g/${siteKey.name.replaceAll('/', ':')}/${navigationNode.uri}`;
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue]:
```suggestion
        this.parentNavigationNodeUrl = `/portal/g/${siteKey.name.replaceAll('/', ':')}/${navigationNode.uri}`;
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue [hunk 9]
```diff
@@ -132,22 +137,52 @@ export default {
       visible: true,
       scheduleVisibility: false,
       nodeType: 'Group',
-      rules: {
+      InitialUrl: '',
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue]:
```suggestion
       parentNavigationNodeUrl: '',
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue [hunk 10]
```diff
@@ -132,22 +137,52 @@ export default {
...
+        }
+      ],
     };
   },
   created() {
     this.$root.$on('open-site-navigation-add-node-drawer', this.open);
   },
+  watch: {
+    nodeId() {
+      this.nodeUrl = `${this.InitialUrl}/${this.nodeId}`;
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue]:
```suggestion
      this.nodeUrl = `{{ $t('siteNavigation.label.nodeId.description') }} ${this.parentNavigationUrl}/${this.nodeId}`;
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue [hunk 11]
```diff
@@ -32,15 +32,20 @@
...
+            @blur="blurOnNodeLabel" />
         </v-card-text>
         <v-card-text class="d-flex flex-grow-1 pb-2">
           <v-label>
             <span class="text-color font-weight-bold mr-6">
               {{ $t('siteNavigation.label.nodeId.title') }} *              
             </span>
+            <p v-if="nodeId && nodeId.length"
+              class="caption text-break">
+              {{ $t('siteNavigation.label.nodeId.description') }} {{ nodeUrl }}
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddNodeDrawer.vue]:
```suggestion
              {{ nodeUrl }}
```

**Review by azayati — CHANGES_REQUESTED** [review]

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**